src/rust/src/conversion/mod.rs

```diff
@@ -356,10 +357,46 @@ impl TryFrom<&str> for Wrap<WindowMapping> {
             "explode" => WindowMapping::Explode,
             v => {
                 return Err(format!(
-                "`mapping_strategy` must be one of ('group_to_rows', 'join', 'explode'), got '{v}'",
+                "`mapping_strategy` must be one of 'group_to_rows', 'join', 'explode'. Got '{v}'",
             ))
             }
         };
         Ok(Wrap(parsed))
     }
 }
+
+impl TryFrom<&str> for Wrap<SetOperation> {
+    type Error = String;
+
+    fn try_from(operation: &str) -> Result<Self, String> {
+        let parsed = match operation {
+            "union" => SetOperation::Union,
+            "intersection" => SetOperation::Intersection,
+            "difference" => SetOperation::Difference,
+            "symmetric_difference" => SetOperation::SymmetricDifference,
+            v => {
+                return Err(format!(
+                    "`operation` must be one of 'union', 'intersection', 'difference', 'symmetric_difference'. Got '{v}'",
+                ))
+            }
```

Comment:
This should be done on the R side with `arg_match()` instead, so it could maybe be `unreachable!()` here?

Reply:
I don't think we need to use the unreachable macro, but I don't think we need to pretty the comment (perhaps something like `unexpected input` or something is good enough.)